harvester_e2e_tests/integration/test_vm_functions.py

```diff
@@ -101,14 +153,414 @@ def test_minimal_vm(api_client, image_id, unique_vm_name, wait_timeout):
+            code, data = api_client.vms.delete(unique_name_for_vm)
+            assert 200 == code, (code, data)
+
+            spec = api_client.vms.Spec.from_dict(data)
+
+            while endtime > datetime.now():
+                code, data = api_client.vms.get(unique_name_for_vm)
+                if 404 == code:
+                    break
+                sleep(3)
+            else:
+                raise AssertionError(
+                    f"Failed to Delete VM({unique_name_for_vm}) with errors:\n"
+                    f"Status({code}): {data}"
+                )
+
+            fails, check = [], dict()
+            for vol in spec.volumes:
+                vol_name = vol['volume']['persistentVolumeClaim']['claimName']
+                check[vol_name] = api_client.volumes.delete(vol_name)
+
+            while endtime > datetime.now():
+                l_check = dict()
+                for vol_name, (code, data) in check.items():
+                    if 200 != code:
+                        fails.append(
+                            (vol_name, f"Failed to delete\nStatus({code}): {data}"))
+                    else:
+                        code, data = api_client.volumes.get(vol_name)
+                        if 404 != code:
+                            l_check[vol_name] = (code, data)
+                check = l_check
+                if not check:
+                    break
+                sleep(5)
+            else:
+                for vol_name, (code, data) in check.items():
+                    fails.append(
+                        (vol_name, f"Failed to delete\nStatus({code}): {data}"))
+
+            assert not fails, (
+                f"Failed to delete VM({unique_vm_name})'s volumes with errors:\n"
+                "\n".join(f"Volume({n}): {r}" for n, r in fails)
+            )
+            break
+        sleep(3)
+    else:
+        raise AssertionError(
+            f"Failed to create VM({overall_vm_obj.get('cpu')} core, \n"
+            f"{overall_vm_obj.get('mem')} RAM, \n"
+            f"{overall_vm_obj.get('disk')} DISK) with errors:\n"
+            f"Phase: {data.get('status', {}).get('phase')}\t"
+            f"Status: {data.get('status')}\n"
+            f"API Status({code}): {data}"
+        )
+
+
+@ pytest.mark.p0
+@ pytest.mark.virtualmachines
+@ pytest.mark.parametrize("machine_types", [("pc", "q35"), ("q35", "pc")],
+                          ids=['pc_to_q35', 'q35_to_pc'])
+def test_update_vm_machine_type(api_client, image_id, unique_vm_name, wait_timeout, machine_types):
+    """Create a VM with machine type then update to another
+
+    Prerequisite:
+        Setting opensuse-image-url set to a valid URL for
+        an opensuse image.
+
+    Manual Test Doc(s):
+        - https://harvester.github.io/tests/manual/virtual-machines/create-new-vm-with-a-machine-type-pc/ # noqa
+        - https://harvester.github.io/tests/manual/virtual-machines/create-new-vm-with-a-machine-type-pc/ # noqa
```

Comment:
duplicate line, it looks we should reference another document that machine type is q35.

Reply:
ahh, yes, duplicate indeed :sweat_smile:  - good catch :smile:  - I've gone ahead and updated the secondary link to point to the appropriate document for q35 :smile: